pkg/reconciler/tenancy/clusterworkspace/deletion/status_condition_util.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2021 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deletion
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/third_party/conditions/apis/conditions/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+)
+
+// NamespaceConditionUpdater interface that translates namespace deleter errors
+// into namespace status conditions.
+type WorkspaceConditionUpdater interface {
+	ProcessDiscoverResourcesErr(e error)
+	ProcessGroupVersionErr(e error)
+	ProcessDeleteContentErr(e error)
+	Update(*tenancyv1alpha1.ClusterWorkspace) bool
+}
```

Comment:
can you follow the patterns for updating conditions in pkg/reconciler/pis/apibindings? This code here feels different in style.

Reply:
updated